crates/uv-git-types/src/oid.rs

```diff
@@ -40,17 +45,17 @@ impl FromStr for GitOid {
             return Err(OidParseError::Empty);
         }
 
-        if s.len() > 40 {
-            return Err(OidParseError::TooLong);
+        if s.len() != 40 {
+            return Err(OidParseError::WrongLength);
         }
 
-        let mut out = [0; 40];
-        out[..s.len()].copy_from_slice(s.as_bytes());
+        if !s.chars().all(|ch| ch.is_ascii_hexdigit()) {
+            return Err(OidParseError::NotHex);
+        }
```

Comment:
I've added strict `GitOid` parsing to this PR, since it simplifies the new code in `sources.rs` and it doesn't seem to break anything.